### ISSUE TITLE
Box regression loss

### ISSUE THREAD

**Issue by dkoguciuk** [google/automl#75]
Hi @mingxingtan,

In the paper there is a statement: " **We also employ commonly-used focal loss**" and in Focal Loss paper there is written: "**The training loss is the sum the focal loss and the standard smooth L1 loss used for box regression**". So I've expected smooth L1 loss for box regression but in the code, there is Huber loss used:

```
  box_loss = tf.losses.huber_loss(
      box_targets,
      box_outputs,
      weights=mask,
      delta=delta,
      reduction=tf.losses.Reduction.SUM)
```

I understand both losses are dependent (smooth L1 loss is equal to huber_loss / delta to be exact), but why exactly did you decide to use Huber instead of L1 like original focal loss paper says? What's the intuition behind it?

**Comment by dkoguciuk** [comment]
Oh, and second question: why do you multiply the normalizer by 4?

```
  normalizer = num_positives * 4.0
  mask = tf.not_equal(box_targets, 0.0)
  box_loss = tf.losses.huber_loss(
      box_targets,
      box_outputs,
      weights=mask,
      delta=delta,
      reduction=tf.losses.Reduction.SUM)
  box_loss /= normalizer
```

**Comment by mingxingtan** [comment]
Hi @dkoguciuk ,  since my main focus is on network architecture, I copied the same loss function from RetinaNet here:

https://github.com/tensorflow/tpu/blob/e2711d5de47f8522af6639dd218a769edef62144/models/official/retinanet/retinanet_model.py#L177

The original RetinaNet paper does mention huber_loss, but I am not sure which one should be ideal.

**Comment by mingxingtan** [comment]
I hope my reply has answered your question, and I am closing this issue for now. Feel free to reopen it if you have further questions. Thanks!
